src/EditorFeatures/Core/SpeculativeEdits/RoslynSpeculativeEditProvider.cs

```diff
@@ -30,39 +32,28 @@ internal sealed class RoslynSpeculativeEditProvider(
         if (document is null)
             return null;
 
-        var documentId = document.Id;
-
         // Clone the existing text into a new editor snapshot/buffer that we can fork independently of the original.
         var clonedSnapshotBeforeEdits = this.CloneWithEdits(options);
 
-        // Grab the ITextSnapshot of this cloned buffer before making any changes.  The SpeculativeEdit api needs it
-        // as part of the returned set of values.
-        var clonedBuffer = clonedSnapshotBeforeEdits.TextBuffer;
-
         // Now create a forked solution that takes the original document and updates it to point at the current state
         // of the text buffer with the edits applied.  Ensure that this properly updates linked files as well so everything
         // is consistent.
-        var newSolution = document.Project.Solution.WithDocumentText(
-            document.Project.Solution.GetRelatedDocumentIds(documentId),
-            clonedBuffer.AsTextContainer().CurrentText,
-            PreservationMode.PreserveIdentity);
+        var clonedBuffer = clonedSnapshotBeforeEdits.TextBuffer;
+        var textContainer = clonedBuffer.AsTextContainer();
 
         // Now, create a preview workspace with that forked document opened within it so that we can lightup features properly there.
```

Comment:
maybe this is more precise.

```suggestion
        // Now, create a preview workspace with an opened forked document within it so that we can lightup features properly there.
```